core/optaplanner-core-impl/src/main/java/org/optaplanner/core/config/heuristic/selector/move/generic/list/kopt/KOptListMoveSelectorConfig.java

```diff
@@ -18,6 +24,14 @@ public class KOptListMoveSelectorConfig extends MoveSelectorConfig<KOptListMoveS
     protected Integer minimumK = null;
     protected Integer maximumK = null;
 
+    protected List<Integer> pickedKDistribution;
```

Comment:
This is a strange field, a list of integers. It's non-self-explanatory, and I don't think we do this anywhere else. Maybe replace with some sort of a supplier type, that can be Javadoc'd?

Also, I don't see it used in the config XML, so maybe we don't need to expose it at all?

Reply:
I see your point, but I question its usefulness. It is something that will be used by a miniscule number of users, but it does complicate the config for everyone, increasing the number of options that need to be benchmarked.

We have a long-standing tradition of exposing every option we possibly could, and I'd like to put a stop to that. The paradox of choice is a real thing, and with millions and millions of different possible configurations, the end user will never be happy with the configuration they have. And they will be nudged towards throwing hours and hours away on benchmarking niche options with marginal, questionable value.

In my opinion, this is a very nasty thing to do to our users. Instead, we should take the burden on our shoulders and make well-reasoned choices for them. The Apple way, not the Linux way; in this case there's still the idea of a continuously adjusting probability-driven selector, which would pick moves based on which have been the most beneficial over the most recent steps.

In summation: I'd rather we remove this option and invest time in researching this smart selector.